src/zkevm_specs/evm/execution/block_ctx.py

```diff
@@ -0,0 +1,44 @@
+from ...util.param import N_BYTES_ACCOUNT_ADDRESS, N_BYTES_U64
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+from ...util import FQ
+
+
+def blockctx(instruction: Instruction):
```

Comment:
I have one question about this function.  Currently in the circuits we have one gadget per execution state.  And some execution states handle multiple opcodes.
Will `blockctx` be a single gadget?
If it's a single gadget, I wonder if there's an advantage if we define a BlockContext execution state that handles all the block context opcodes in a single execution state :thinking: Maybe we would reduce the number of constraints a bit?

Reply:
yeah, in circuit code there is a  `blockctxGadget` for BlockContext ops.  `define a BlockContext execution state` is a good idea for reducing constraints. i will refactor it latter.

Thank you very much for your careful reviews.